File: Sorting/01_Selection_Sort/cpp/SelectionSort.cpp
#include<iostream>
using namespace std;

void selectionSort(int arr[], int n) {
	for (int i = 0; i < n - 1; i++) {
		int imin = i;
		for (int j = i + 1; j < n; j++) {
			if (arr[j] < arr[imin]) {
				imin = j;
			}
		}
		int temp = arr[imin];
		arr[imin] = arr[i];
		arr[i] = temp;
	}
}


int main() {
	int arr[] = { 2, 7, 4, 1, 5, 3 };
	int sizeOfArray = sizeof(arr) / sizeof(int);

	cout << "unsorted array: ";
	for (int i = 0; i < sizeOfArray; i++) {
		cout << arr[i] << " ";
	}

	selectionSort(arr, sizeOfArray);

	cout << "\nsorted array: ";
	for (int i = 0; i < sizeOfArray; i++) {
		cout << arr[i] << " ";
	}
}

File: Sorting/01_Selection_Sort/python/selectionSort.py
def selectionSort(lst):
    length_of_list = len(lst)
    for i in range(length_of_list - 1):
        imin = i
        for j in range(i + 1, length_of_list):
            if lst[j] < lst[imin]:
                imin = j
        temp = lst[i]
        lst[i] = lst[imin]
        lst[imin] = temp
    return


if __name__ == '__main__':
    unsorted_list = [2, 7, 4, 1, 5, 3]
    print('unsorted list is: {}'.format(unsorted_list))
    selectionSort(unsorted_list)
    print('sorted list is: {}'.format(unsorted_list))
